### PR TITLE
Patch former commit to fix compile_commands.json generation

### LINKED ISSUE #9895: make compile_commands.json fix
### Description of the enhancement or error report
The command `make compile_commands.json` may fail when a large number of source files are present. This was observed when all the modules were included in an `MooseApp`'s build.

### Rationale for the enhancement or information for reproducing the error
A fix can be applied that fixes this issue in `framework/moose.mk`, but it requires Make v4+
Another workaround is to customize an app's Makefile to include only the needed modules

### Identified impact
Immediate fix for users with Make v4+. No impact for other users.

### PR DESCRIPTION
**Pull request by BruceWyxv**
closes #9895 (again)

### CODE REVIEW

File: framework/moose.mk [hunk 1]
```diff
@@ -219,17 +219,17 @@ clobberall:: clobber
...
-	else
+else
 		@echo $(CURDIR) > .compile_commands.json
 		@echo $(libmesh_CXX) >> .compile_commands.json
 		@echo $(libmesh_CPPFLAGS) $(libmesh_CXXFLAGS) $(app_INCLUDES) $(libmesh_INCLUDE) $(ADDITIONAL_INCLUDES) >> .compile_commands.json
 		@echo $(compile_commands_all_srcfiles) >> .compile_commands.json
-	endif
+endif
 	@ $(FRAMEWORK_DIR)/scripts/compile_commands.py < .compile_commands.json > compile_commands.json
 	@rm .compile_commands.json
```
Comment (dschwen) [inline framework/moose.mk]: Cahn you please double check that these cleanup statements after the `endif`, which belong to the `compile_commands.json::` target are indented correctly?

Comment (dschwen) [reply framework/moose.mk]: Looks like they are though

Comment (BruceWyxv) [reply framework/moose.mk]: I have updated the indentation of everything to match. They seem to be cosmetic changes only, though.
The cleanup commands do run during the `compile_commands.json::` recipe. I also verified that they are not hit during the regular build recipe.

**Review by dschwen — APPROVED** [review]
Alrighty then...

**Pull request merged**